server/src/main/java/org/elasticsearch/action/ActionModule.java

```diff
@@ -767,6 +771,11 @@ public <Request extends ActionRequest, Response extends ActionResponse> void reg
         actions.register(UpdateHealthInfoCacheAction.INSTANCE, UpdateHealthInfoCacheAction.TransportAction.class);
         actions.register(FetchHealthInfoCacheAction.INSTANCE, FetchHealthInfoCacheAction.TransportAction.class);
 
+        // Synonyms
+        if (SynonymsAPI.isEnabled()) {
```

Comment:
A new SynonymsAPI class for doing the FF check has been created - it didn't seem right to depend on the whole SynonymsManagementService to do just that

Reply:
I am not clear what dependency you talking about. As `SynonymsManagementService` is already part of `server`, we don't introduce any extra depedency?

Reply:
Thanks, Carlos, makes sense to remove this class when FF is removed.